.config-schema.json

```diff
@@ -430,6 +430,35 @@
                     "x-env-variable": "OPENFGA_CHECK_QUERY_CACHE_TTL"
                 }
             }
+        },
+        "dispatchThrottling": {
+            "type": "object",
+            "properties": {
+                "enabled": {
+                    "description": "enable throttling when request's number of dispatches is high",
+                    "type": "bool",
+                    "default": "false",
+                    "x-env-variable": "OPENFGA_DISPATCH_THROTTLING_ENABLED"
+                },
+                "timeTickerFrequency": {
+                    "description": "defines how frequent dispatch throttling will be evaluated",
+                    "type": "duration",
+                    "default": "10µs",
+                    "x-env-variable": "OPENFGA_DISPATCH_THROTTLING_TIME_TICKER_FREQUENCY"
+                },
+                "level": {
+                    "description": "define the number of dispatches to be throttled",
+                    "type": "integer",
+                    "default": "100",
+                    "x-env-variable": "OPENFGA_DISPATCH_THROTTLING_LEVEL"
+                },
+                "rate": {
```

Comment:
What's the difference between the `timeTickerFrequency` and the `rate`? Aren't they derivatives of one another? e.g. if I configure a rate of 10 dispatches/sec then the ticker interval is defined as 1/10. The ticker frequency is always the inverse of the rate.

Shouldn't the rate be an exponentially decaying function of the delta between the `level` and the current observed dispatch_count (e.g. `(X - dispatch.level)*exp`)?

Reply:
timeTickerFrequency = how frequent time.Ticker will run to evaluate dispatch
rate = how many ticks to launch a job

Maybe we use a different word to represent "rate"?

>Shouldn't the rate be an exponentially decaying function of the delta between the level and the current observed dispatch_count (e.g. (X - dispatch.level)*exp)?

No, there is no notion of exponential decay.  We have removed all exponential decay when we remove the multi-level design.

Reply:
> timeTickerFrequency = how frequent time.Ticker will run to evaluate dispatch
rate = how many ticks to launch a job

So if `timeTickerFrequency=1ms` and `rate=10` then we'll dispatch at a rate of `10ms`. What I'm saying is that we don't need 2 configs to define that. `timeTickerFrequency` defines the unit, rate defines the value. Instead, how about we just define `rate=10ms` and/or maybe a better name is `interval=10ms`.

> No, there is no notion of exponential decay. We have removed all exponential decay when we remove the multi-level design.

This is my bigger concern - what is stopping us from doing expo decay without the multi-level design, that's not clear to me? The delay is a function of the delta between the dispatch threshold and the current dispatch count multiplied by some exponential.

Reply:
suggestion: for starters, the only two config knobs we should expose is enabled/level. Let's keep the defaults for the rate/interval in code for now.

Reply:
updated to remove rate and have the frequency instead.